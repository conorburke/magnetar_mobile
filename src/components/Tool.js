import React, { Component } from 'react';
import { Text, TouchableOpacity, View } from 'react-native';
import { connect } from 'react-redux';
import Icon from 'react-native-vector-icons/Ionicons';

import * as actions from '../actions';

class Tool extends Component {
  openToolDetail() {
    console.log('tools details props', this.props.tool);
    this.props.navigator.push({
      screen: 'seker.ToolDetailScreen',
      title: 'Tool Detail',
      backButtonTitle: 'Back to Tools',
      passProps: this.props.tool
    });
    // Alert.alert('Tapped Text!');
  }

  handleDelete(id) {
    alert(`id to delete is ${id}`);
    this.props.deleteTool(id);
  }

  render() {
    console.log('propssss', this.props);
    console.log('navigator', this.props.navigator);
    return (
      <View style={styles.containerStyle}>
        <TouchableOpacity>
          <Text onPress={this.openToolDetail.bind(this)}>{this.props.tool.Title}</Text>
        </TouchableOpacity>
        <TouchableOpacity>
          <Icon onPress={() => this.handleDelete(this.props.tool.ID)} size={30} name='ios-trash' color='red'/>
        </TouchableOpacity>
      </View>
      
    )
  }
}

const styles = {
  containerStyle: {
    borderBottomWidth: 1,
    padding: 5,
    backgroundColor: '#FFF',
    justifyContent: 'space-between',
    flexDirection: 'row',
    borderColor: '#DDD',
    position: 'relative',
    width: '75%'
  }
};

export default connect(null, actions)(Tool);