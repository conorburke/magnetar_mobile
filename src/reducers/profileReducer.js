import { CREATE_PROFILE } from '../actions/types';

export default function(state = {}, action) {
    // console.log(action);
    switch (action.type) {
        case CREATE_PROFILE:
            return action.payload || state;
        default:
            return state
    }
}