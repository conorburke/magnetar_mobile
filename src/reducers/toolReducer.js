import { FETCH_TOOL } from '../actions/types';

export default function(state = [], action) {
    // console.log(action);
    switch (action.type) {
        case FETCH_TOOL:
            // console.log('action.payload', action.payload);
            // console.log('state', state);
            return action.payload || state;
            // return action.payload || false;
        default:
            return state
    }
}