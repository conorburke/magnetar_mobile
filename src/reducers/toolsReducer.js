import { FETCH_TOOLS } from '../actions/types';

export default function(state = [], action) {
    // console.log(action);
    switch (action.type) {
        case FETCH_TOOLS:
            // console.log('action.payload', action.payload);
            // console.log('state', state);
            return action.payload || state;
            // return action.payload || false;
        default:
            return state
    }
}