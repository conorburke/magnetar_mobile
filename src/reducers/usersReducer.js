import { FETCH_USERS, FILTER_USERS, SET_PHONE_NUMBER } from '../actions/types';

const initialState = {
  usersList: [],
  usersSearch: '',
  phoneNumber: ''
};

export default function(state = initialState, action) {
  switch (action.type) {
    case FETCH_USERS:
      return { ...state, ...{ usersList: action.payload } };
    case FILTER_USERS:
      return { ...state, ...{ usersSearch: action.payload } };
    case SET_PHONE_NUMBER:
      return { ...state, ...{ phoneNumber: action.payload } };
    default:
      return state;
  }
}
