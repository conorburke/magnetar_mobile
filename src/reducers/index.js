import { combineReducers } from 'redux';

import authReducer from './authReducer'
import toolsReducer from './toolsReducer';
import usersReducer from './usersReducer';

export default combineReducers({
    auth: authReducer,
    tools: toolsReducer,
    users: usersReducer
});