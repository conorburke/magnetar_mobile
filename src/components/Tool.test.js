import React from 'react';
import { shallow } from 'enzyme';
import Tool from './Tool';
import { createStore } from 'redux';
import reducers from '../reducers';

const store = createStore(reducers);

const tool = { Price: 1, ToolType: 'Hammer', FirstName: 'Randy' };

const wrapper = shallow(<Tool tool={tool} store={store} />).dive();

it('should render the Tool component', () => {
  expect(wrapper).toMatchSnapshot();
});
