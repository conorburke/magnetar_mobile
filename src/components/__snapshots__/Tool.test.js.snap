// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the Tool component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Tool
    authUser={[Function]}
    createProfile={[Function]}
    deleteTool={[Function]}
    fetchProfile={[Function]}
    fetchTool={[Function]}
    fetchTools={[Function]}
    fetchUserTools={[Function]}
    fetchUsers={[Function]}
    filterTools={[Function]}
    filterUsers={[Function]}
    profile={Object {}}
    setPhoneNumber={[Function]}
    store={
      Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "@@observable": [Function],
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
    tool={
      Object {
        "FirstName": "Randy",
        "Price": 1,
        "ToolType": "Hammer",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#F5F5F5",
              "fontSize": 25,
            }
          }
        />
        <Component
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#F5F5F5",
                "fontSize": 15,
              }
            }
          >
            Cost: $ 1
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#F5F5F5",
                "fontSize": 15,
              }
            }
          >
            Type: Hammer
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#F5F5F5",
                "fontSize": 15,
              }
            }
          >
            Owner: Randy undefined
          </Text>
        </Component>
      </TouchableOpacity>,
      "style": Object {
        "backgroundColor": "#666666",
        "borderBottomWidth": 1,
        "borderColor": "#DDD",
        "borderRadius": 10,
        "flexDirection": "row",
        "margin": 5,
        "padding": 5,
        "position": "relative",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "activeOpacity": 0.2,
        "children": Array [
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#F5F5F5",
                "fontSize": 25,
              }
            }
          />,
          <Component
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#F5F5F5",
                  "fontSize": 15,
                }
              }
            >
              Cost: $ 1
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#F5F5F5",
                  "fontSize": 15,
                }
              }
            >
              Type: Hammer
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#F5F5F5",
                  "fontSize": 15,
                }
              }
            >
              Owner: Randy undefined
            </Text>
          </Component>,
        ],
        "onPress": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": undefined,
            "ellipsizeMode": "tail",
            "style": Object {
              "color": "#F5F5F5",
              "fontSize": 25,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#F5F5F5",
                    "fontSize": 15,
                  }
                }
              >
                Cost: $ 1
              </Text>,
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#F5F5F5",
                    "fontSize": 15,
                  }
                }
              >
                Type: Hammer
              </Text>,
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#F5F5F5",
                    "fontSize": 15,
                  }
                }
              >
                Owner: Randy undefined
              </Text>,
            ],
            "style": Object {
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "Cost: $ 1",
                "ellipsizeMode": "tail",
                "style": Object {
                  "color": "#F5F5F5",
                  "fontSize": 15,
                },
              },
              "ref": null,
              "rendered": "Cost: $ 1",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "Type: Hammer",
                "ellipsizeMode": "tail",
                "style": Object {
                  "color": "#F5F5F5",
                  "fontSize": 15,
                },
              },
              "ref": null,
              "rendered": "Type: Hammer",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "Owner: Randy undefined",
                "ellipsizeMode": "tail",
                "style": Object {
                  "color": "#F5F5F5",
                  "fontSize": 15,
                },
              },
              "ref": null,
              "rendered": "Owner: Randy undefined",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#F5F5F5",
                "fontSize": 25,
              }
            }
          />
          <Component
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#F5F5F5",
                  "fontSize": 15,
                }
              }
            >
              Cost: $ 1
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#F5F5F5",
                  "fontSize": 15,
                }
              }
            >
              Type: Hammer
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#F5F5F5",
                  "fontSize": 15,
                }
              }
            >
              Owner: Randy undefined
            </Text>
          </Component>
        </TouchableOpacity>,
        "style": Object {
          "backgroundColor": "#666666",
          "borderBottomWidth": 1,
          "borderColor": "#DDD",
          "borderRadius": 10,
          "flexDirection": "row",
          "margin": 5,
          "padding": 5,
          "position": "relative",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "activeOpacity": 0.2,
          "children": Array [
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#F5F5F5",
                  "fontSize": 25,
                }
              }
            />,
            <Component
              style={
                Object {
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#F5F5F5",
                    "fontSize": 15,
                  }
                }
              >
                Cost: $ 1
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#F5F5F5",
                    "fontSize": 15,
                  }
                }
              >
                Type: Hammer
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#F5F5F5",
                    "fontSize": 15,
                  }
                }
              >
                Owner: Randy undefined
              </Text>
            </Component>,
          ],
          "onPress": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": undefined,
              "ellipsizeMode": "tail",
              "style": Object {
                "color": "#F5F5F5",
                "fontSize": 25,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#F5F5F5",
                      "fontSize": 15,
                    }
                  }
                >
                  Cost: $ 1
                </Text>,
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#F5F5F5",
                      "fontSize": 15,
                    }
                  }
                >
                  Type: Hammer
                </Text>,
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#F5F5F5",
                      "fontSize": 15,
                    }
                  }
                >
                  Owner: Randy undefined
                </Text>,
              ],
              "style": Object {
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "space-between",
              },
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "accessible": true,
                  "allowFontScaling": true,
                  "children": "Cost: $ 1",
                  "ellipsizeMode": "tail",
                  "style": Object {
                    "color": "#F5F5F5",
                    "fontSize": 15,
                  },
                },
                "ref": null,
                "rendered": "Cost: $ 1",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "accessible": true,
                  "allowFontScaling": true,
                  "children": "Type: Hammer",
                  "ellipsizeMode": "tail",
                  "style": Object {
                    "color": "#F5F5F5",
                    "fontSize": 15,
                  },
                },
                "ref": null,
                "rendered": "Type: Hammer",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "accessible": true,
                  "allowFontScaling": true,
                  "children": "Owner: Randy undefined",
                  "ellipsizeMode": "tail",
                  "style": Object {
                    "color": "#F5F5F5",
                    "fontSize": 15,
                  },
                },
                "ref": null,
                "rendered": "Owner: Randy undefined",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
